src/interface/src/app/services/plan.service.ts

```diff
@@ -587,7 +598,7 @@ export class PlanService {
       all: {
         ...currentState.all,
       },
-      currentScenarioId: scenarioId,
+      currentScenarioId: Number(scenarioId),
```

Comment:
Can we change the signature of  `updateStateWithScenario` to use number instead of string so we avoid casting?

Reply:
Reverted the scenario type to have a string id instead of a number since it seems more straightforward to keep it as a string (had changed it to a number to bridge some gaps with switching from Project (which had a number id), but ended up being able to just switch everything to string)